lib/product-import.coffee

```diff
@@ -224,7 +224,7 @@ class ProductImport
 
   _errorLogger: (res, logger) =>
     if @_summary.failed < @errorLimit or @errorLimit is 0
-      logger.error(util.inspect(res, false, 3, true), "Skipping product due to an error")
+      logger.error(util.inspect(res, false, 10, true), "Skipping product due to an error")
```

Comment:
- Why would we limit to any level at all? I am not sure if it is required to use inspect here
- In your screenshot i see that `Skipping product due to an error` is logged as the first statement. If you look at example i posted [here](https://github.com/sphereio/sphere-product-import/issues/144#issuecomment-470624236) then it is logged in the end which is quite unintuitive to read. Also i see in my output many weird characters like `statusCode: \u001b[33m400\u001b[39m,\n     message:\n      \u001b[32m'The value`. What logger did you use for the screenshot output? Could you please try it with Bunyan too?

Reply:
- The limit can be helpfull when printing circular objects.
 - I took the screenshot from tests and it was started with `npm run tests | bunyan` command.
 - Regarding those weird characters - these are [color codes](https://askubuntu.com/a/681719) used to change text color in terminal. From where did you take that error? From AWS logs or somewhere else?

Also you can define your own error logger and pass it as an [optional parameter](https://github.com/sphereio/sphere-product-import/blob/master/lib/product-import.coffee#L25).

Reply:
Adding some more info and thoughts
- We added `util.inspect` here as the default is just 2 levels deep. So we had problems debugging things properly in Impex. (It uses the [sphere-node-cli](https://github.com/sphereio/sphere-node-cli) which in turn uses the [winston logger](https://github.com/winstonjs/winston).)
- I'd be very wary of including all levels to `Infinity`. It might lead to completely enormous error outputs.
- Maybe the better long-term solution is to do like you suggested, @butenkor, and make things completely customizable by also allowing a callback or a simple `logLevel` variable?
- I'm not sure what is going on with the breaking of formatting 😕

Reply:
> The limit can be helpful when printing circular objects.

Do you expect circular objects? If yes then we have different issue here to solve anyway. I would like to avoid future requests to extend from 10 to 20...

> Regarding those weird characters - these are color codes used to change text color in the terminal. From where did you take that error? From AWS logs or somewhere else?

It is written into log file like that. It is also being collected that way from SDTOUT by GCP Stackdriver

> Also, you can define your own error logger and pass it as an optional parameter

This is what we do currently

Reply:
Also when i pipe file content to bunyan
`tail +1f my-cli.log | bunyan -l 50`
then i do not see color chars anymore but `Skipping product due to an error` is always in the end of logged error.

Reply:
> Maybe the better long-term solution is to do like you suggested, @butenkor, and make things completely customizable by also allowing a callback or a simple logLevel variable

Considering that we use different loggers and have different requirements on the output limitations it seems like optional callback `errorCallback` would be preferred solution here.

Reply:
maybe the broken output is because Bunyan does not handle these "color codes" .. @butenkor please check this comment - https://github.com/trentm/node-bunyan/issues/513#issuecomment-339147137

Reply:
Maybe usage of `BUNYAN_NO_COLOR` as env var might help but i do not have color issues when the error is not treated with `inspect` tool. Also it does not address the issue why custom message appears in the end of the log. WDYT about `errorCallback`?

Reply:
It is possible to set a global depth level using `util.inspect.defaultOptions.depth = 10` property but that would mean that it will change depth also for modules which will include and use `sphere-product-import` as a dependency.

The `errorCallback` option is already available but I believe that the proper logging should be implemented here so it won't have to be done in all projects separately.

Reply:
> util.inspect(res, false, 10, false /** instead of true **/)

Maybe this is an option but i would go with level 20 at least. Thanks for reminding me that we have errorCallback already. I will try to use now in our project.

Reply:
updated here ca04d60

Reply:
Thank You!